go/apps/api/openapi/spec/common/roleReference.yaml

```
@@ -0,0 +1,29 @@
+    pattern: "^[a-zA-Z][a-zA-Z0-9_-]*$"
+    description: |
+      Identifies the role by its human-readable name within the workspace.
+      Role names must start with a letter and contain only letters, numbers, underscores, or hyphens.
+      Names must be unique within the workspace and are case-sensitive.
+      More readable than IDs but vulnerable to integration breaks if roles are renamed.
+      Use IDs for automation and names for human-configured integrations.
+    example: admin
+additionalProperties: false
+example:
+  id: role_1234567890abcdef
+  name: admin
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)